okreplay-gradle-plugin/src/main/kotlin/okreplay/OkReplayPlugin.kt

```diff
@@ -1,77 +1,67 @@
 package okreplay
 
-import com.android.build.gradle.*
+import com.android.build.gradle.AppExtension
+import com.android.build.gradle.BaseExtension
+import com.android.build.gradle.LibraryExtension
 import com.android.build.gradle.api.BaseVariant
+import com.android.build.gradle.internal.api.TestedVariant
+import org.gradle.api.DomainObjectSet
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.Task
-import org.gradle.api.UnknownTaskException
-import org.gradle.api.internal.DefaultDomainObjectSet
-import javax.inject.Inject
 
-class OkReplayPlugin
-@Inject constructor() : Plugin<Project> {
-  private lateinit var project: Project
+class OkReplayPlugin : Plugin<Project> {
 
   override fun apply(project: Project) {
-    this.project = project
-    if (project.plugins.hasPlugin(AppPlugin::class.java)
-        || project.plugins.hasPlugin(LibraryPlugin::class.java)) {
-      applyPlugin()
-    } else {
-      throw IllegalArgumentException("OkReplay plugin couldn't be applied. "
-          + "The Android or Library plugin must be configured first.")
+    project.plugins.withId("com.android.application") {
```

Comment:
why not  like this?
```kotlin
withType(AppPlugin::class.java)
```

Reply:
In general I prefer using ids because they are documented but since we are hard-coding extension types anyway I see that there is not much point in this.